.github/workflows/deploy.yml

```diff
@@ -0,0 +1,60 @@
+name: Deployment Workflow
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Setup Environment
+      run: |
+        pip install jinja2
+        generate/generate-python.py --output-dir=brping
+        git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+        git fetch origin deployment
+        git config user.email "support@bluerobotics.com"
+        git config user.name "BlueRobotics-CI"
+
+    - name: install and test
+      run: |
+        echo "installing package..."
+        pip install . --user
+
+        echo "testing message api..."
+        python brping/pingmessage.py
+
+
+    - name: Generate and Commit Files
+      run: |
+        cat ci/deploy-whitelist | xargs git add -f
+        git commit -m "temporary-commit" || true
+        git checkout deployment
+        rm -rf *
+        git checkout HEAD@{1} ci/deploy-whitelist
+        cat ci/deploy-whitelist | xargs git checkout HEAD@{1}
+        git --no-pager diff --staged
+        git rm -f ci/deploy-whitelist
```

Comment:
why not just calling deploy.sh over replacing it ?

Reply:
it gets so small without the overhead that I didn't see the point in keeping it. but I'll that. I think it makes things more... traceable